core/src/main/java/org/apache/accumulo/core/iterators/Combiner.java

```diff
@@ -313,4 +392,48 @@ public static void setColumns(IteratorSetting is, List<IteratorSetting.Column> c
   public static void setCombineAllColumns(IteratorSetting is, boolean combineAllColumns) {
     is.addOption(ALL_OPTION, Boolean.toString(combineAllColumns));
   }
+
+  /**
+   * @since 1.6.4 1.7.1 1.8.0
+   */
+  public static enum DeleteHandlingAction {
+    /**
+     * Do nothing when a a delete is observed by a combiner during a major compaction.
+     */
+    IGNORE,
+
+    /**
+     * Log an error when a a delete is observed by a combiner during a major compaction. An error is not logged for each delete entry seen. Once a
+     * combiner has seen a delete during a major compaction and logged an error, it will not do so again for at least an hour.
+     */
+    LOG_ERROR,
+
+    /**
+     * Pass all data through during partial major compactions, no reducing is done. With this option reducing is only done during scan and full major
+     * compactions, when deletes can be correctly handled.
+     */
+    REDUCE_ON_FULL_COMPACTION_ONLY
+  }
+
+  /**
+   * Combiners may not work correctly with deletes. Sometimes when Accumulo compacts the files in a tablet, it only compacts a subset of the files. If a delete
+   * marker exists in one of the files that is not being compacted, then data that should be deleted may be combined. See
+   * <a href="https://issues.apache.org/jira/browse/ACCUMULO-2232">ACCUMULO-2232</a> for more information.
+   *
+   * <p>
+   * This method allows users to configure how they want to handle the combination of delete markers, combiners, and major compactions. The default behavior is
+   * {@link DeleteHandlingAction#LOG_ERROR}. See the javadoc on each {@link DeleteHandlingAction} enum for a description of each option.
+   *
+   * <p>
+   * For correctness deletes should not be used with columns that are combined OR the {@link DeleteHandlingAction#REDUCE_ON_FULL_COMPACTION_ONLY} option should
+   * be used. Only reducing on full major compactions may have negative performance implications.
+   *
```

Comment:
I don't think this should be coupled with this fix.  Also not sure about doing this in 1.6.  I opened a new issue [ACCUMULO-4011](https://issues.apache.org/jira/browse/ACCUMULO-4011)